src/pretalx/submission/models/submission.py

```diff
@@ -288,38 +297,71 @@ def _set_state(self, new_state, force=False, person=None):
+                # logging.debug(
+                #     'available_count_old: {}'.format(available_count_old))
+                # logging.debug(
+                #     'self.available_count {}'.format(self.available_count))
+                # range_to_delete = list(range(
+                #     self.available_count, available_count_old))
+                # logging.debug('range_to_delete: {}'.format(range_to_delete))
+                qs_delete = TalkSlot.objects.filter(
+                    submission=self,
+                    schedule=self.event.wip_schedule,
+                    available_index__in=range_to_delete,
+                )
+                # logging.debug('qs_delete: {}'.format(
+                #     [i.available_index for i in qs_delete]))
+                qs_delete.delete()
+            # logging.debug('TalkSlot update_or_create')
+            for index in range(self.available_count):
+                TalkSlot.objects.update_or_create(
+                    submission=self,
+                    schedule=self.event.wip_schedule,
+                    defaults={'is_visible': True},
+                    available_index=index,
+                )
+        else:
+            TalkSlot.objects.filter(
+                submission=self, schedule=self.event.wip_schedule
+            ).delete()
+        # debug: print all now available Talks
+        qs_temp = TalkSlot.objects.filter(
+            submission=self,
+            schedule=self.event.wip_schedule,
+        )
+        logging.debug('TalkSlots new: {}'.format(
+            [i.available_index for i in qs_temp]))
 
-        TalkSlot.objects.filter(
-            submission=self, schedule=self.event.wip_schedule
-        ).delete()
+    def make_submitted(self, person=None, force=False, orga=False):
+        self._set_state(SubmissionStates.SUBMITTED, force, person=person)
+        self.update_TalkSlots()
 
     def confirm(self, person=None, force=False, orga=False):
         self._set_state(SubmissionStates.CONFIRMED, force, person=person)
         self.log_action('pretalx.submission.confirm', person=person, orga=orga)
         self.slots.filter(schedule=self.event.wip_schedule).update(is_visible=True)
-        from pretalx.schedule.models import TalkSlot
-
-        TalkSlot.objects.update_or_create(
-            submission=self,
-            schedule=self.event.wip_schedule,
-            defaults={'is_visible': True},
-        )
+        self.update_TalkSlots()
 
     def accept(self, person=None, force=False, orga=True):
         previous = self.state
         self._set_state(SubmissionStates.ACCEPTED, force, person=person)
         self.log_action('pretalx.submission.accept', person=person, orga=True)
-
-        from pretalx.schedule.models import TalkSlot
-
-        TalkSlot.objects.update_or_create(
-            submission=self,
-            schedule=self.event.wip_schedule,
-            defaults={'is_visible': True},
-        )
+        self.update_TalkSlots()
```

Comment:
Please call `update_talk_slots` from `_set_state`, not from each individual state change method.

Reply:
done